eis_toolkit/vector_processing/extract_shared_lines.py

```diff
@@ -0,0 +1,49 @@
+import geopandas as gpd
+from beartype import beartype
+
+from eis_toolkit import exceptions
+
+
+def _extract_shared_lines(polygons: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    shared_lines_list = []
+
+    # Used to make sure that same polygon couples are not examined together multiple times
+    reversed_couples = set()
+
+    for i, poly1 in polygons.iterrows():
+        for j, poly2 in polygons.iterrows():
```

Comment:
It is an option to use itertools.combinations to replace the for loops with just one. Then we would also not need to compare i to j.